src/foam/comics/DAOController.js

```diff
@@ -89,34 +89,78 @@ foam.CLASS({
       `,
       factory: function() { return this.NullBorder.create(); }
     },
-    {
-      class: 'Boolean',
-      name: 'filtersEnabled',
-      documentation: `Set to true if you want to completely hide the search
-        panel and the button to toggle it.`,
-      value: true
-    },
     {
       class: 'Boolean',
       name: 'searchHidden',
       documentation: `Used internally to keep track of whether the search panel
         is currently hidden or not.`,
       value: false
     },
+    {
+      class: 'Enum',
+      of: 'foam.comics.SearchMode',
+      name: 'searchMode',
+      documentation: `
+        The level of search capabilities that the controller should have.
+      `,
+      factory: function() {
+        return this.SearchMode.FULL;
+      },
+      postSet: function(oldValue, newValue) {
+        switch ( newValue ) {
+          case this.SearchMode.FULL:
+            this.PREDICATE.view = { class: 'foam.u2.view.ReciprocalSearch' };
```

Comment:
Would it be simpler to make the view used be a function of the searchMode than to do this? Also, PREDICATE.view is a global variable, so if you had more than one DAOController active at once (like in a dashboard or Relationships in an object which contain nested DAOControllers, then this will leak into the other controllers.

Reply:
Resolved by e45125a.